### ISSUE TITLE
NFS Most Wanted launch fails

### ISSUE THREAD

**Issue by mradermaxlol** [iXit/Mesa-3D#192]
Hello!  
The issue: NFS Most Wanted (back from 2005) crashes (on launch) immediately when Native D3D9 is enabled. I've tried both 32- and 64-bit prefixes, neither works. I'm attaching some logs from console, as well as my xorg config from xorg.conf.d.  
If it matters, i'm using wine from my [overlay](https://github.com/mradermaxlol/wine-d3d9-gentoo) on Gentoo with ~amd64. Note that staging features were disabled at the moment of my testing.  
**Attachments:**
[winelog.txt](https://github.com/iXit/Mesa-3D/files/143211/winelog.txt)
[xorgconf.txt](https://github.com/iXit/Mesa-3D/files/143213/xorgconf.txt)
[xorglog.txt](https://github.com/iXit/Mesa-3D/files/143212/xorglog.txt)

**Comment by mradermaxlol** [comment]
> Unhandled exception: page fault on read access to 0x00000000 in 32-bit code (0x7d543545).
This line makes me think that there's something wrong with the 32-bit part of the code.

**Comment by axeldavy** [comment]
Unfortunately your logs are not very useful to debug this issue.

You can get a useful log with NINE_DEBUG=all, but it will display things only if mesa was built with --enable-debug

**Comment by siro20** [comment]
I got the same crash, but it crashes with vanilla wine, too.
It's likely not related to d3d9 at all.

**Comment by mradermaxlol** [comment]
siro, maybe you are just using 64-bit prefix or your wine is broken. For me it worked on vanilla 1.9.3/1.9.4 and staging (same vers) on both 32- and 64-bit prefixes.  
Are you getting EXACTLY the same error?  
  
Oh, by the way, I will post debug logs once I've made my new ebuild work and have recompiled Mesa with debug options. Maybe today, maybe tomorrow.

**Comment by pontostroy** [comment]
wine 1.9.4 staging+nine
NFS MW works for me with nine and wine-csmt
some options does not available with csmt (shadow detail) 
and with nine and shadow detail enabled game looks very weird, too dark and smoke textures is broken.
https://www.youtube.com/watch?v=BkapgkohAuM

**Comment by siro20** [comment]
I tested with Ixit/wine (wine 1.9.2) and 32bit prefix.
I got a null pointer dereference in the application itself.
Maybe my wine is broken, I'll try 1.9.3 and 1.9.4.

**Comment by mradermaxlol** [comment]
@pontostroy works great for me with CSMT, too, and Shadow Detail is also unavalible. Everything works flawlessly even with UniWS-patched exe (my display is 16:9 1080p). But, well, I'm still trying to make Nine work there, and also (as you know) GCC 5.3.0 compilator has issues with Wine 1.9.4, and I'm working on it.

**Comment by mradermaxlol** [comment]
@axeldavy here you go. This one is from my current mesa, if it's not what you are looking for - tell me and I will provide you with logs with mesa compiled with **debug** USE flag.  

[winelog.txt](https://github.com/iXit/Mesa-3D/files/146677/winelog.txt)

**Comment by pontostroy** [comment]
tested with docker image based on Tumbleweed (gcc 5.3.1) and still works fine, even with UniWS-patched exe

**Comment by mradermaxlol** [comment]
@pontostroy hmm... Then it's probably ebuilds' issue, I don't see anything else. I will try some people to test it, but no promises.  
Oh, maybe it's all about extra patches from your source package? I will experiment with them, maybe it helps.

**Comment by mradermaxlol** [comment]
Hmm... May it be happening because of absent DLL redirects? Like, from d3d9.dll to d3d9-nine.dll?

**Comment by pontostroy** [comment]
no, i have the same redirect

**Comment by mradermaxlol** [comment]
@pontostroy, I have **no** redirects at all, maybe that's it?

**Comment by mradermaxlol** [comment]
@pontostroy could you give me a list of dependencies for your OpenSuSE wine package? Both x86_32 (i686) and x86_64 (amd64), if you please ;) If think the issue is there. If so, I'll provide my overlay with a sample config for portage with neccessary packages mentioned.

**Comment by pontostroy** [comment]
@mradermaxlol 
http://pastebin.com/ARcq0EN7

**Comment by mradermaxlol** [comment]
@pontostroy hmm, everything seems fine. I've actually done some testing with Mirror's Edge, and Nine runs flawlessly there. I'm thinking it's actually a game's issue... IDK.

**Comment by axeldavy** [comment]
Can we close ?

**Comment by mradermaxlol** [comment]
@axeldavy I don't think so. Actually, Nine works with Mirror's Edge for me, but doesn't with neither STALKER CoP nor NFSMW. Should I open another issue with STALKER or write it here? Anyway, I'll publish some Nine logs with Mirror's Edge and STALKER, as I've broken my NFS installation recently.

**Comment by mradermaxlol** [comment]
[Mirror's Edge log](https://github.com/iXit/Mesa-3D/files/182257/melog.txt)  
[S.T.A.L.K.E.R. CoP 1.6.0.2 log](https://github.com/iXit/Mesa-3D/files/182259/stlog.txt)  
I'm on vanilla Wine 1.9.6 with Nine patch applied. I managed to play the whole level in ME on high-ultra settings and I got a crash with S.T.A.L.K.E.R. X-Ray was also complaining after the crash.

**Comment by axeldavy** [comment]
With which flags are you building mesa ? Try to use -O0, etc.

**Comment by mradermaxlol** [comment]
@axeldavy of course I'm building 32-bit, and all my prefixes are 32-bit. I'll try experimenting with -Oxxx flags, okay.

**Comment by mradermaxlol** [comment]
Alright, as Mesa 11.2 got released, I'm gonna test stuff with it. I'll report here ASAP.

**Comment by mradermaxlol** [comment]
Umm... Nothing new, really. Well, except of WineD3D being broken for now XD (but that's probably Xorg regression). I'll try to run the game with Nine using Gears On Gallium, maybe I'm just ass-handed?
